GiniPayBank/Classes/Core/GiniPayBank.swift

```diff
@@ -6,33 +6,86 @@
-     the user will be presented with a digital representation of their invoice where they
-     can see individual line items and are able to amend them or choose to not to pay for them.
+     Fetches the payment request via payment request id.
+
+     - parameter paymentRequestId: Id of payment request.
+     - parameter completion: An action for processing asynchronous data received from the service with Result type as a paramater. Result is a value that represents either a success or a failure, including an associated value in each case.
+     Completion block called on main thread.
+     In success returns the payment request structure.
+     In case of failure error from the server side.
+
      */
-    
-    @objc public var returnAssistantEnabled = true
-    
+    public func receivePaymentRequest(paymentRequestId: String, completion: @escaping (Result<PaymentRequest, GiniPayBankError>) -> Void) {
+        paymentService.paymentRequest(id: paymentRequestId) { result in
+            DispatchQueue.main.async {
+                switch result {
+                case let .success(paymentRequest):
+                    self.paymentRequesterScheme = paymentRequest.requesterURI 
+                    completion(.success(paymentRequest))
+                case let .failure(error):
+                    completion(.failure(.apiError(error)))
+                }
+            }
+        }
+    }
+
     /**
-     Singleton to make configuration internally accessible in all classes of the Gini PayBank SDK.
+     Resolves the payment via payment request id.
+
+     - parameter paymentRequesId: Id of payment request.
+     - parameter completion: An action for processing asynchronous data received from the service with Result type as a paramater. Result is a value that represents either a success or a failure, including an associated value in each case.
+     Completion block called on main thread.
+     In success returns the payment structure.
+     In case of failure error from the server side.
+
      */
-   public static var shared = GiniPayBank()
-    
-   public override init() {}
+    public func resolvePayment(paymentRequesId: String, paymentInfo: PaymentInfo, completion: @escaping (Result<Payment, GiniPayBankError>) -> Void) {
+        paymentService.resolvePaymentRequest(id: paymentRequesId, recipient: paymentInfo.recipient, iban: paymentInfo.iban, amount: paymentInfo.amount, purpose: paymentInfo.purpose, completion: { result in
+            DispatchQueue.main.async {
+                switch result {
+                case .success: break
+                case let .failure(error):
+                    completion(.failure(.apiError(error)))
+                }
+            }
+        }
+        )
+    }
 
+    /**
+     Returns back to the business app.
+     Return button should be active for the user after resolving the payment request.
+
+     */
+    public func returnBackToBusinessAppHandler() {
```

Comment:
In the Android version this function gets the `PaymentRequest` as a parameter and then gets the `requestURI` from it so we could avoid storing the URI in this class.

Reply:
this part will be redesigned : requestURI will be returned from resolve payment , but I'm totally agree to remove this var in class

Reply:
Oh, right. I forgot.